### PR TITLE
Add federalist pipeline

### PR DESCRIPTION
**Pull request by hursey013**
## Changes proposed in this pull request:
- Add federalist-pipeline.yml config for federalist-garden-build pipeline to deploy federalist-garden-build app to production
- Moved `set-pipeline-((deploy-env))` job to top of file as it is elsewhere

## security considerations
Needed for SCR

### CODE REVIEW

File: ci/federalist-pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    - do: *test
+
+    on_failure:
+      put: src
+      resource: pr-main
+      params:
+        path: src
+        status: failure
+        base_context: concourse
+        context: test
```
Comment (apburnes) [inline ci/federalist-pipeline.yml:76]: Update `context` to something like `test-federalist` so we don't conflict with pages production PR check.

File: ci/federalist-pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+
+    on_success:
+      put: src
+      resource: pr-main
+      params:
+        path: src
+        status: success
+        base_context: concourse
+        context: test
```
Comment (apburnes) [inline ci/federalist-pipeline.yml:85]: Update `context` to something like `test-federalist` so we don't conflict with pages production PR check.

File: ci/federalist-pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    type: slack-notification
+    source:
+      url: ((slack-webhook-url))
+
+  - name: gh-status
+    type: cogito
+    check_every: 1h
+    source:
+      owner: 18F
+      repo: federalist-garden-build
```
Comment (apburnes) [inline ci/federalist-pipeline.yml:212]: We may want to turn this into a credhub variable to make our GH org migration simpler.

File: ci/pipeline.yml [hunk 4]
```diff
@@ -41,6 +41,16 @@ test: &test
...
 
+  - name: set-pipeline-((deploy-env))
+    serial: true
+    plan:
+      - get: src
+        resource: src-((deploy-env))
+        params: {depth: 1}
+        trigger: true
+      - set_pipeline: build-container
+        file: src/ci/pipeline.yml
```
Comment (apburnes) [inline ci/pipeline.yml:52]: Do we need to add instance variables here for staging and prod?

**Review by apburnes — COMMENTED** [review]
Just a few comments on pipeline params.

File: ci/pipeline.yml (hunk 4 above)
Comment (hursey013) [reply ci/pipeline.yml:52]: Added!

**Review by apburnes — APPROVED** [review]

**Pull request merged**